Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1457,12 +1458,21 @@ private void SetMaterials()
                 handleGrabbedMaterial.SetFloatArray("_InnerGlowColor", color);
             }
         }
-        private void InitializeDataStructures()
+
+        private void InitializeRigRoot()
         {
-            rigRoot = new GameObject(rigRootName).transform;
+            var rigRootObj = new GameObject(rigRootName);
+            rigRoot = rigRootObj.transform;
             rigRoot.parent = transform;
 
+            var pH = rigRootObj.AddComponent<PointerHandler>();
+            pH.OnPointerDown.AddListener(OnPointerDown);
+            pH.OnPointerDragged.AddListener(OnPointerDragged);
```

Comment:
@Railboy I believe adding a listener to onpointerdragged, which gets called every frame, may increase memory usage. This is based on reading this comment in #5156 

> UnityEvents are toxic. Eg, ManipulationHandler allocates about 1.5k when invoking any ManipulationEvent, 1.1k of which is UnityEvent overhead. ManipulationEvents are discrete so they're not a huge problem. But we should avoid using these for update events.

I am thinking to create a follow up PR to remove this line, since we don't even do anything in OnPointerDragged (my mistake that I added it). Are there any other changes that you see in this PR that could cause lots of memory usage? I was only worried about onpointerdragged because it gets called every frame.

Reply:
_UnityEvents are toxic_ is a very strong statement that moves us away from the most Unity-like way of handling events. I'd say we need more hard data before accepting that as a general guideline. I'd like to see what's the allocation footprint of a invoking a UnityEvent in a Master build, that's what I'd use as a reference.

Reply:
@luis-valverde-ms Yes that was a bit hyperbolic, I was just stunned to see those events allocating more than 1k per invocation.

>I'd like to see what's the allocation footprint of a invoking a UnityEvent in a Master build, that's what I'd use as a reference.

Good idea, I'll give that a shot and we'll see what happens.

Reply:
Thank you! Let me know how it goes. I'd be suprised if UnityEvent is doing something very unreasonable perf/allocation wise in Master builds.